lib/trento/domain/cluster/cluster.ex

```diff
@@ -472,6 +487,18 @@ defmodule Trento.Domain.Cluster do
     }
   end
 
+  defp maybe_emit_cluster_health_changed_event(%Cluster{
+         cluster_id: cluster_id,
+         discovered_health: discovered_health,
+         selected_checks: selected_checks,
+         health: health
+       })
+       when selected_checks == [] do
+    if discovered_health != health do
```

Comment:
guard is not needed here.
Also, we could maybe refactor this by creating a function called `maybe_add_checks_health(checks_health, selected_cheacks)`
to pipe it in the `new_health` match in the function below to remove repetition.

```elixir
    new_health =
      [discovered_health]
      |> maybe_add_checks_health(checks_health, selected_checks)
      |> Enum.filter(& &1)
      |> HealthService.compute_aggregated_health()
```